sklearn/tests/test_metaestimators.py

```diff
@@ -43,7 +44,11 @@ def __init__(self, name, construct, skip_methods=(),
     DelegatorData('BaggingClassifier', BaggingClassifier,
                   skip_methods=['transform', 'inverse_transform', 'score',
                                 'predict_proba', 'predict_log_proba',
-                                'predict'])
+                                'predict']),
+    DelegatorData('SelfTraining', lambda est: SelfTraining(est),
+                  skip_methods=['transform', 'inverse_transform', 'score',
```

Comment:
Why are these skipped?

Reply:
I'm skipping `transform` and `inverse_transform` because, to my understanding, they aren't part of the estimator API (and therefore I don't expose them as methods). 

`predict_proba` is skipped because the SelfTrainingClassifier throws an error if it is missing.

`predict_log_proba` and `predict` are no longer skipped.

Reply:
Concerning `score`, I think there might be inconsistency in the test. The score method I use comes straight from `ClassifierMixin`: with the following signature
`def score(self, X, y, sample_weight=None):`

However the test calls score with only `X`, which causes it to fail. I suspect that the reason for this is that since `Pipeline` allows score to be called with `y=None`, the test uses the following score method for the estimator:
`def score(self, X, *args, **kwargs):`

This causes the test to fail on my classifier.

Reply:
In general, I'm a bit confused as to what the `score` method is supposed to do, since some classifiers (for example `ClassifierMixin`, `RFE`, `EllipticEnvelope`) return the accuracy score on the passed `X` and `y`.

However, other estimators, like `PCA` and `KernelDensity` use their own version of score that completely ignores `y` and only uses `X`.

Reply:
Yes, score can be a little confusing. But PCA and KernelDensity are not supervised so their evaluation is not by default against a ground truth. They report model likelihoods instead.

Yes, it might be a mistake in the test to not try passing y to score.

Reply:
I've made a PR to fix the test. Once it is merged I will be able to remove `score` from the skipped methods.